contracts/marketplace/src/execute.rs

```diff
@@ -462,6 +462,20 @@ pub fn execute_fund_renewal(
     let payment = must_pay(&info, NATIVE_DENOM)?;
 
     let mut ask = asks().load(deps.storage, ask_key(token_id))?;
+
+    // get the renewal price
+    let renewal_price =
+        query_ask_renew_price(deps.as_ref(), ask.renewal_time, (&token_id).to_string())?;
+
+    // make sure that we do not over fund the renewal
+    // based on the price we got back
+    ensure!(
```

Comment:
It should be possible to do this safely without the unchecked unwraps.

Reply:
addressed in [7c61fd9](https://github.com/public-awesome/names/pull/262/commits/7c61fd9137ee14c716f903a8eb999ddf873b0d6c)

Reply:
@shanev is this a reasonable approach?